conclave-common/src/test/kotlin/com/r3/conclave/common/EnclaveConstraintTest.kt

```diff
@@ -278,6 +280,54 @@ class EnclaveConstraintTest {
         }
     }
 
+    @Test
+    fun `check against security level - secure enclave`() {
+        val secureEnclave = TestEnclaveInstanceInfo(summary = SECURE)
+
+        listOf(STALE, SECURE).forEach { securityLevelRequiredByClient ->
+            val constraint = EnclaveConstraint.parse("SEC:$securityLevelRequiredByClient C:${secureEnclave.enclaveInfo.codeHash}")
+            assertDoesNotThrow { constraint.check(secureEnclave) }
+        }
+
+        assertThrows<InvalidEnclaveException> {
+            val securityLevelRequiredByClient = INSECURE
+            val constraint = EnclaveConstraint.parse("SEC:$securityLevelRequiredByClient C:${secureEnclave.enclaveInfo.codeHash}")
+            constraint.check(secureEnclave)
+        }
+    }
+
+    @Test
+    fun `check against security level - stale enclave`() {
+        val secureEnclave = TestEnclaveInstanceInfo(summary = STALE)
+
+        assertDoesNotThrow {
+            val securityLevelRequiredByClient = STALE
+            val constraint = EnclaveConstraint.parse("SEC:$securityLevelRequiredByClient C:${secureEnclave.enclaveInfo.codeHash}")
+            constraint.check(secureEnclave)
+        }
+
+        listOf(INSECURE, SECURE).forEach { securityLevelRequiredByClient ->
+            val constraint = EnclaveConstraint.parse("SEC:$securityLevelRequiredByClient C:${secureEnclave.enclaveInfo.codeHash}")
+            assertThrows<InvalidEnclaveException> { constraint.check(secureEnclave) }
+        }
+    }
```

Comment:
Same here, in case of enclave's security level = STALE, then STALE and SECURE shouldn't throw an exception. Only INSECURE should throw an exception.